bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/EntryLocationIndex.java

```diff
@@ -204,46 +202,21 @@ public void removeOffsetFromDeletedLedgers() throws IOException {
                 firstKeyWrapper.set(ledgerId, 0);
                 lastKeyWrapper.set(ledgerId, Long.MAX_VALUE);
 
-                Entry<byte[], byte[]> firstKeyRes = locationsDb.getCeil(firstKeyWrapper.array);
-                if (firstKeyRes == null || ArrayUtil.getLong(firstKeyRes.getKey(), 0) != ledgerId) {
-                    // No entries found for ledger
-                    log.info("No entries found for ledger {}", ledgerId);
-                    continue;
-                }
-
-                long firstEntryId = ArrayUtil.getLong(firstKeyRes.getKey(), 8);
-                long lastEntryId = getLastEntryInLedgerInternal(ledgerId);
-                log.info("Deleting index for ledger {} entries ({} -> {})",
-                        new Object[] { ledgerId, firstEntryId, lastEntryId });
+                batch.deleteRange(firstKeyWrapper.array, lastKeyWrapper.array);
```

Comment:
The problem is that getting that information can be very expensive and this is one of the reasons for this change. 

To get the first and last entry for a ledger in the index, we need to use rocksdb iterator. These iterators are very fast in normal conditions, but they degrade a lot when there are many deleted keys as they generate a lot of random reads.